assets/src/components/SurveyModal.js

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import Button from '@material-ui/core/Button';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import IconButton from '@material-ui/core/IconButton';
+import Modal from '@material-ui/core/Modal';
+import CloseIcon from '@material-ui/icons/Close';
+
+function getModalStyle() {
+  const top = 10;
+  const left = 50;
+
+  return {
+    top: `${top}%`,
+    left: `${left}%`,
+    transform: `translate(-${left}%)`,
+  };
+}
+
+const useStyles = makeStyles((theme) => ({
+  paper: {
+    position: 'absolute',
+    width: 400,
+    backgroundColor: theme.palette.background.paper,
+    border: '2px solid #000',
+    boxShadow: theme.shadows[5],
+    padding: theme.spacing(2, 4, 3),
+  },
+  dialogTitle: {
+    display: 'flex',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    padding: '0px'
+  },
+  iframeContainer: {
+    position: 'relative',
+    overflow: 'hidden',
+    width: '100%',
+    paddingTop: '150%'
+  },
+  /* Then style the iframe to fit in the container div with full height and width */
+  iframe: {
+    position: 'absolute',
+    top: 0,
+    left: 0,
+    bottom: 0,
+    right: 0,
+    width: '100%',
+    height: '100%'
+  }
+}));
+
+export default function SurveyModal(props) {
+  const classes = useStyles();
+  // getModalStyle is not a pure function, we roll the style only on the first render
+  const [modalStyle] = React.useState(getModalStyle);
```

Comment:
So you need to run this on first render because of the iframe and absolute/relative positioning, correct? I'm still not super comfortable with this pattern, so I'm making sure I understand.

Is there anything stopping you from just using `const modalStyle = getModalStyle()`? That worked for me. If we can avoid the state variable (it only gets set once), that seems ideal.

Reply:
Actually, I investigated further, and I'm not seeing (but could still easily be missing something) why we can't just create a new class to use with `useStyles`. I added the following object to the style array, and changed that opening body `div` tag to look like: `<div className={`${classes.paper} ${classes.modal}`}>`. That seems to not break anything?

`  modal: {
    top: '10%',
    left: '50%',
    transform: 'translate(-50%)'
  }

Reply:
Sure I think that's fine. I think I had it like this because I was playing around a lot with the style positioning here and the variables to get it positioned well for mobile and desktop. But just hardcoding it would work.

Thanks for the review!